core/trie_test.go

```diff
@@ -207,17 +166,21 @@ func TestTriePut(t *testing.T) {
 			root:  nil,
 		},
 	}
-	for idx, test := range tests {
-		if err := trie.Put(test.key, test.value); err != nil {
-			t.Errorf("TestTriePut: Put() failed at test #%d", idx)
-		}
-		if value, err := trie.Get(test.key); err != nil || !value.Equal(test.value) {
-			t.Errorf("TestTriePut: Get() failed at test #%d", idx)
-		}
-		if test.root != nil && !test.root.Equal(trie.root) {
-			t.Errorf("TestTriePut: Unexpected root at test #%d", idx)
+	DoOnTempTrie(251, func(trie *Trie) error {
+		for idx, test := range tests {
```

Comment:
For each test we should create a subtest, to make it easier to identify which cases passed/failed. We can do this in a separate PR.